settings/src/components/numeric-control.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Input field for values that use digits, but aren't strictly numbers in the mathematical sense.
+ *
+ * The classic example is a credit card, but in our context we have TOTP codes, backup codes, etc. We may want to
+ * display them with spaces for easier reading, etc.
+ *
+ * If we used Gutenberg's `NumberControl`, we'd have to hide the extraneous UI elements, and it would still be
+ * using the underlying `input[type="number"]`, which has some accessibility issues.
+ *
+ * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#accessibility
+ * @link https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
+ * @link https://stackoverflow.com/a/66759105/450127
+ */
+export default function NumericControl( props ) {
+	return (
+		<input
+			{ ...props }
+			type="text"
+			inputMode="numeric"
+			autoComplete={ props.autoComplete || 'off' }
+			pattern={ props.pattern || "[0-9 ]*" }
```

Comment:
Consider adding a title to explain the pattern:

`title="Verification codes only include numbers"`

Gives:
<img width="357" alt="Screen Shot 2566-03-10 at 13 54 36" src="https://user-images.githubusercontent.com/1657336/224226791-cd68de8b-36b4-47e0-bad8-d49d1410c499.png">

Reply:
Great catch! I added that in fd77cf3